src/stan/lang/function_signatures.h

```diff
@@ -38,6 +38,9 @@ add("add", MATRIX_T, DOUBLE_T, MATRIX_T);
 for (size_t i = 0; i < base_types.size(); ++i) {
   add("add", base_types[i], base_types[i]);
 }
+for (size_t i = 0; i < base_types.size(); ++i)
+  add("append_array", expr_type(base_types[i], 1), expr_type(base_types[i], 1),
```

Comment:
I think this can be generalized from just the base types to a broad set of types.